core/camel-support/src/main/java/org/apache/camel/support/cache/DefaultProducerCache.java

```diff
@@ -54,6 +54,10 @@
 public class DefaultProducerCache extends ServiceSupport implements ProducerCache {
 
     private static final Logger LOG = LoggerFactory.getLogger(DefaultProducerCache.class);
+    private static final RejectedExecutionException REJECTED_EXECUTION_EXCEPTION
+            = new RejectedExecutionException("CamelContext is stopped");
+    private volatile boolean filledStackTrace;
```

Comment:
You could rather use an `AtomicReference<RejectedExecutionException>` instead , then in `send` method you lazily create it with `exchange.setException(REJECTED_EXECUTION_EXCEPTION.updateAndGet(ex -> ex == null ? new RejectedExecutionException("CamelContext is stopped") : ex));`

Reply:
🤔

Reply:
I'm rolling back on this one for now (it's a minor thing - not important for the hot path), but I liked the idea. In the future, we should explore this in other areas of the code base.

Reply:
Yes that could be interesting in a place where we could create a new exception very frequently as building a stack trace is very slow